server/src/main/java/org/apache/druid/metadata/segment/CachedSegmentMetadataTransaction.java

```diff
@@ -74,14 +77,18 @@ class CachedSegmentMetadataTransaction implements SegmentMetadataTransaction
     if (leaderSelector.isLeader()) {
       this.startTerm = leaderSelector.localTerm();
     } else {
-      throw InternalServerError.exception("Not leader anymore. Cannot start transaction.");
+      throw DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.SERVICE_UNAVAILABLE)
+                          .build("Not leader anymore. Cannot start transaction.");
     }
   }
 
   private void verifyStillLeaderWithSameTerm()
   {
     if (!isLeaderWithSameTerm()) {
-      throw InternalServerError.exception("Not leader anymore. Failing transaction.");
+      throw DruidException.forPersona(DruidException.Persona.USER)
+                          .ofCategory(DruidException.Category.SERVICE_UNAVAILABLE)
+                          .build("Not leader anymore. Failing transaction.");
```

Comment:
This message isn't great for a user-facing error. The error message should help the user figure out what to do next. Better would be like: "This API is not currently available. Please try your request again."

Reply:
Updated.